schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,47 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: The name of the column
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
```

Comment:
Not that this should drive the ASDF design, but I don't think `astropy.table.Table` currently supports a display name, so we'd need to add support for that (or shove it in `meta`).

@taldcroft: Do you have any thoughts on adding a display name to Table columns?  I doing this would be most useful if `Columns` understood them and used them in the `repr` etc.  Though I could definitely see the downside to this in terms of confusion (having two, possibly very different, names for the same thing).   I think I'm personally -0 on the idea because of that, but I could be convinced otherwise.